cli/src/utils.rs

```diff
@@ -80,24 +80,28 @@ impl PlatformSelector {
                 // defaults based on previous selections if previous selections are not empty
                 // otherwise, selects all
                 let all_ios_archs = IosArch::all_strings();
-                let defaults: Vec<bool> = if config.ios.is_empty() {
+                let defaults: Vec<bool> = if config.ios.is_none() {
```

Comment:
We still need `is_empty` here. The hashset object exsits (aka. `is_some()`) still can be an empty one.

Reply:
the same as `android` below

Reply:
I decided not to let it default select all
so when it's empty or it's len = 0
none of them will be selected
[029aaec](https://github.com/zkmopro/mopro/pull/417/commits/029aaec3858a5897fbbc747d4efea1752807973e)

Reply:
cuious why you decided it? I think selecting all in default is nice. BTW, `is_none` and `is_empty` is different.

Reply:
1. it takes time to build
2. some of them are not commonly used nowadays (e.g. `x86_64-apple-ios`, all the other android targets except for `aarch64-linux-android`)
3. some proving systems don't support all targets
    e.g. Noir, `rapidsnark`/`witnesscalc`